Regression in v0.9.4: undefined selector Time

I found a regression in Yaegi, it appeared in v0.9.4, it affects the plugin `plugin-simplecache`:

```
error: /tmp/pilot-gop211396919/src/github.com/traefik/plugin-simplecache/cache.go:12:2: import \"github.com/pquerna/cachecontrol\"
error: /tmp/pilot-gop211396919/src/github.com/traefik/plugin-simplecache/vendor/github.com/pquerna/cachecontrol/api.go:21:2: import \"github.com/pquerna/cachecontrol/cacheobject\"
error: /tmp/pilot-gop211396919/src/github.com/traefik/plugin-simplecache/vendor/github.com/pquerna/cachecontrol/cacheobject/object.go:255:32: undefined selector Time"
```

https://github.com/pquerna/cachecontrol/blob/master/cacheobject/object.go

https://github.com/traefik/yaegi/compare/v0.9.3...v0.9.4

related to https://github.com/traefik/plugin-simplecache/issues/12

<!-- 
The following program `sample.go` triggers a panic:

```go
package main

func main() {
// add a sample
}
```

Expected result:
```console
$ go run ./sample.go
// output
```

Got:
```console
$ yaegi ./sample.go
// output
```
-->

I am seeing the same thing when running test on gorilla websockets.

```shell
$ yaegi test -unsafe ./
test: example_test.go:12:2: import "github.com/gorilla/websocket" error: client.go:54:29: undefined selector Conn
```

--- a/_test/time14.go
+++ b/_test/time14.go
@@ -1,0 +1,20 @@
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+var t time.Time
+
+func f() time.Time {
+	time := t
+	return time
+}
+
+func main() {
+	fmt.Println(f())
+}
+
+// Output:
+// 0001-01-01 00:00:00 +0000 UTC

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1381,7 +1381,7 @@ func (interp *Interpreter) cfg(root *node, importPath string) ([]*node, error) {
 			n.val = sc.def
 			for i, c := range n.child {
 				var typ *itype
-				typ, err = nodeType(interp, sc, returnSig.child[1].fieldType(i))
+				typ, err = nodeType(interp, global(sc), returnSig.child[1].fieldType(i))
 				if err != nil {
 					return
 				}

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1381,7 +1381,7 @@ func (interp *Interpreter) cfg(root *node, importPath string) ([]*node, error) {
 			n.val = sc.def
 			for i, c := range n.child {
 				var typ *itype
-				typ, err = nodeType(interp, global(sc), returnSig.child[1].fieldType(i))
+				typ, err = nodeType(interp, sc.upperLevel(), returnSig.child[1].fieldType(i))
 				if err != nil {
 					return
 				}

--- a/interp/scope.go
+++ b/interp/scope.go
@@ -113,9 +113,9 @@ func (s *scope) pop() *scope {
 	return s.anc
 }
 
-// global returns the global scope from the current one.
-func global(s *scope) *scope {
-	for s != nil && !s.global {
+func (s *scope) upperLevel() *scope {
+	level := s.level
+	for s != nil && s.level == level {
 		s = s.anc
 	}
 	return s

--- a/interp/scope.go
+++ b/interp/scope.go
@@ -113,6 +113,14 @@ func (s *scope) pop() *scope {
 	return s.anc
 }
 
+// global returns the global scope from the current one.
+func global(s *scope) *scope {
+	for s != nil && !s.global {
+		s = s.anc
+	}
+	return s
+}
+
 // lookup searches for a symbol in the current scope, and upper ones if not found
 // it returns the symbol, the number of indirections level from the current scope
 // and status (false if no result).

